### ISSUE TITLE
client auth not working on a newly joined node (3.3.10)

### ISSUE THREAD

**Issue by phanhuy1502** [etcd-io/etcd#11348]
(read) Please read https://github.com/etcd-io/etcd/blob/master/Documentation/reporting_bugs.md.

## description
- On an etcd cluster with authentication enabled (by `etcdctl auth enable`), when a new node joins in, requests to that node are rejected because the authentication doesn't work on the new node.
- Issue resolved when authentication is disable and enable again

## version
- etcd: 3.3.10
- os: ubuntu 16.04

## reproduce steps
- Set up a etcd cluster with 3 nodes
- Create root user and enable authentication enable (with `etcdctl auth enable`)
- Add a new node to the cluster (following [this guide](https://github.com/etcd-io/etcd/blob/master/Documentation/op-guide/runtime-configuration.md#add-a-new-member)).
- Send a get request to the newly added node (and only to that node)
```bash
ETCDCTL_API=3 etcdctl --endpoints "http://<new-node-ip>:2379" --user root:pw get '' --prefix --debug
```
- Get request is rejected because the new node fails to authenticate the client
- Issues resolved when authentication is disabled and enabled again (`etcdctl auth disable`, then `etcdctl auth enable`)

## expected behaviour
- new node should work properly without manual intervention (force disabling and enabling auth)

### etcdclt logs

```
ETCDCTL_API=3 etcdctl --endpoints "http://10.128.22.106:2379" --user root:pw get '' --prefix --debug
ETCDCTL_CACERT=
ETCDCTL_CERT=
ETCDCTL_COMMAND_TIMEOUT=5s
ETCDCTL_DEBUG=true
ETCDCTL_DIAL_TIMEOUT=2s
ETCDCTL_DISCOVERY_SRV=
ETCDCTL_ENDPOINTS=[http://10.128.22.106:2379]
ETCDCTL_HEX=false
ETCDCTL_INSECURE_DISCOVERY=true
ETCDCTL_INSECURE_SKIP_TLS_VERIFY=false
ETCDCTL_INSECURE_TRANSPORT=true
ETCDCTL_KEEPALIVE_TIME=2s
ETCDCTL_KEEPALIVE_TIMEOUT=6s
ETCDCTL_KEY=
ETCDCTL_USER=root:pw
ETCDCTL_WRITE_OUT=simple
INFO: 2019/11/12 16:31:26 dialing to target with scheme: ""
INFO: 2019/11/12 16:31:26 could not get resolver for scheme: ""
INFO: 2019/11/12 16:31:26 ccBalancerWrapper: updating state and picker called by balancer: IDLE, 0xc4201a1320
INFO: 2019/11/12 16:31:26 dialing to target with scheme: ""
INFO: 2019/11/12 16:31:26 could not get resolver for scheme: ""
INFO: 2019/11/12 16:31:26 balancerWrapper: is pickfirst: false
INFO: 2019/11/12 16:31:26 balancerWrapper: got update addr from Notify: [{10.128.22.106:2379 <nil>}]
INFO: 2019/11/12 16:31:26 ccBalancerWrapper: new subconn: [{10.128.22.106:2379 0  <nil>}]
INFO: 2019/11/12 16:31:26 balancerWrapper: handle subconn state change: 0xc4202f0ad0, CONNECTING
INFO: 2019/11/12 16:31:26 ccBalancerWrapper: updating state and picker called by balancer: CONNECTING, 0xc4201a1320
INFO: 2019/11/12 16:31:26 balancerWrapper: handle subconn state change: 0xc4202f0b20, CONNECTING
INFO: 2019/11/12 16:31:26 ccBalancerWrapper: updating state and picker called by balancer: CONNECTING, 0xc4201a1320
INFO: 2019/11/12 16:31:26 balancerWrapper: handle subconn state change: 0xc4202f0ad0, READY
INFO: 2019/11/12 16:31:26 clientv3/balancer: pin "10.128.22.106:2379"
INFO: 2019/11/12 16:31:26 ccBalancerWrapper: updating state and picker called by balancer: READY, 0xc4201a1320
INFO: 2019/11/12 16:31:26 balancerWrapper: got update addr from Notify: [{10.128.22.106:2379 <nil>}]
INFO: 2019/11/12 16:31:26 ccBalancerWrapper: removing subconn
INFO: 2019/11/12 16:31:26 balancerWrapper: handle subconn state change: 0xc4202f0b20, SHUTDOWN
INFO: 2019/11/12 16:31:26 ccBalancerWrapper: updating state and picker called by balancer: READY, 0xc4201a1320
WARNING: 2019/11/12 16:31:26 Failed to dial 10.70.12.101:2379: grpc: the connection is closing; please retry.
WARNING: 2019/11/12 16:31:26 Failed to dial 10.70.12.101:2379: grpc: the connection is closing; please retry.
INFO: 2019/11/12 16:31:26 clientv3/retry: error "rpc error: code = Unauthenticated desc = etcdserver: invalid auth token" on pinned endpoint "10.128.22.106:2379"
INFO: 2019/11/12 16:31:26 clientv3/auth-retry: error "rpc error: code = Unauthenticated desc = etcdserver: invalid auth token" on pinned endpoint "10.128.22.106:2379"
INFO: 2019/11/12 16:31:26 dialing to target with scheme: ""
INFO: 2019/11/12 16:31:26 could not get resolver for scheme: ""
INFO: 2019/11/12 16:31:26 clientv3/retry: error "rpc error: code = Unauthenticated desc = etcdserver: invalid auth token" on pinned endpoint "10.128.22.106:2379"
INFO: 2019/11/12 16:31:26 clientv3/auth-retry: error "rpc error: code = Unauthenticated desc = etcdserver: invalid auth token" on pinned endpoint "10.128.22.106:2379"
INFO: 2019/11/12 16:31:26 dialing to target with scheme: ""
... 
(repeated logs till context timeout)
```

### logs on the newly joined etcd node

```
2019-11-12 16:31:26.791613 W | auth: invalid auth token: FibhNPsUUtsFgSqI.23974000
2019-11-12 16:31:26.873784 W | auth: invalid auth token: dPIbXRkMTKxpzOEo.23974001
2019-11-12 16:31:26.956780 W | auth: invalid auth token: RnNGbBYvNpayUNsT.23974002
2019-11-12 16:31:27.041495 W | auth: invalid auth token: CFbhRZlwfkGTjoxc.23974003
2019-11-12 16:31:27.125029 W | auth: invalid auth token: YUxKIcqtlEnQOYmj.23974004
2019-11-12 16:31:27.208448 W | auth: invalid auth token: hHNpLjyctDezSUTf.23974005
2019-11-12 16:31:27.294387 W | auth: invalid auth token: ZqNoptVlSfNascAK.23974006
2019-11-12 16:31:27.377547 W | auth: invalid auth token: hJFrgOvHQqcSrnDC.23974007
2019-11-12 16:31:27.461948 W | auth: invalid auth token: ZZRxndcokMFNbIax.23974008
2019-11-12 16:31:27.546650 W | auth: invalid auth token: FrYHDOAqZYfDRdJN.23974009
2019-11-12 16:31:27.629027 W | auth: invalid auth token: weDScDMokVoeRBnZ.23974010
2019-11-12 16:31:27.711757 W | auth: invalid auth token: sgeRTlcFIkoLauol.23974011
2019-11-12 16:31:27.798326 W | auth: invalid auth token: KurCBaXcbwoDmQeH.23974012
2019-11-12 16:31:27.882270 W | auth: invalid auth token: EwfkSMuEdwqFeyuC.23974013
2019-11-12 16:31:27.968030 W | auth: invalid auth token: emRSyjPDygMpKpVf.23974014
2019-11-12 16:31:28.051510 W | auth: invalid auth token: qAdyjweNwMlSFUrs.23974015
2019-11-12 16:31:28.134293 W | auth: invalid auth token: qlkYjCCVyZmUrpXF.23974016
WARNING: 2019/11/12 16:31:28 grpc: Server.processUnaryRPC failed to write status: connection error: desc = "transport is closing"
```

**Comment by phanhuy1502** [comment]
update: this is not always reproducible, i'll update here if can find the setup to reproduce this all the times

**Comment by mitake** [comment]
@phanhuy1502 simple token is just for testing purposes. For example, the newly joined nodes won't copy generated tokens which are valid in the existing nodes. Could you try jwt token? With jwt token this problem won't happen. You can find how to use jwt here: https://github.com/etcd-io/etcd/blob/master/Documentation/op-guide/configuration.md#auth-flags

**Comment by lsnan** [comment]
I had the same problem, my etcd version was 3.5.1

**Comment by spzala** [comment]
Please try newly released 3.5.3 and open a new issue if the problem exist. Thanks!
